lib/_http_outgoing.js

```diff
@@ -277,6 +278,11 @@ OutgoingMessage.prototype.setTimeout = function setTimeout(msecs, callback) {
 // any messages, before ever calling this.  In that case, just skip
 // it, since something else is destroying this connection anyway.
 OutgoingMessage.prototype.destroy = function destroy(error) {
+  if (this.destroyed) {
+    return;
+  }
+  this.destroyed = true;
+
   if (this.socket) {
     this.socket.destroy(error);
   } else {
```

Comment:
I think that lines [286-291](https://github.com/nodejs/node/pull/32148/files#diff-286202fdbdd74ede6f5f5334b6176b5cR286-R291) should be replaced with `this.req.destroy(error);`

Reply:
I thought the same thing. Unfortunately, in the `onSocket` case `this.socket` is never set.

Reply:
> Unfortunately, in the onSocket case this.socket is never set.

What's wrong with this?

Reply:
> What's wrong with this?

Nothing specific, just makes the implementation a little harder to follow.

Reply:
> just makes the implementation a little harder to follow.

Why? I see no reason (but I may be mistaken)... What's the problem here?

Reply:
in other places (e.g. net, fs) it usually goes like this:

```js
prototype._destroy = function (err, cb) {
  if (this.socket) {
    this.socket.destroy(err, cb);
  else 
    this.on('socket', () => this._destroy(err, cb));
};
```

which is not possible here since, `'socket'` is not always emitted.

Reply:
There is no problem here really though. It can be refactored later using e.g. https://github.com/nodejs/node/pull/29656

Reply:
> which is not possible here since, 'socket' is not always emitted.

But we would call `this.req.destroy(error)`, not `this.socket.destroy`. The `ClientRequest` instance would handle the destroy.

> There is no problem here really though. It can be refactored later using e.g. #29656

I'd rather prefer to have it done in this PR to avoid mess and confusion.

Reply:
?

Reply:
> But we would call this.req.destroy(error), not this.socket.destroy. The ClientRequest instance would handle the destroy.

Where?

Reply:
https://github.com/nodejs/node/pull/32148/files#diff-e3bc37430eb078ccbafe3aa3b570c91aR350-R368

Reply:
That won’t work since socket is not set on the instance.

Reply:
> That won’t work since socket is not set on the instance.

How so? To emit the `response` event there must be a `socket` set on the ClientRequest instance. I still don't see why can't we just `this.req.destroy(error)`.

Reply:
I am still confused about what exactly you are suggesting... Can you just post a snippet of how you want it to be?

Reply:
Sorry for being unclear. What I mean is:

```js
OutgoingMessage.prototype.destroy = function destroy(error) {
  if (this.destroyed) {
    return;
  }
  this.destroyed = true;


  this.req.destroy(error);
};
```

Reply:
`OutgoingMessage` is `req`. That doesn't make sense. Are you maybe confusing this with `IncomingMessage`?

Reply:
Oh man I'm so sorry. I got so confocused 🤦‍♂️

Reply:
My apologies, you've been right from the beginning. I don't know why I did thinking the way I did, it just makes no sense now :facepalm: